From e735c3ed510af4637a493e336fba68a7705bcf17 Mon Sep 17 00:00:00 2001
From: tonishi92 <tatsuo.onishi@gmail.com>
Date: Wed, 14 Dec 2022 13:08:11 +0100
Subject: [PATCH 70/70] [Debug] A floating invalid error can happen when zcloud
 and/or clfra equal to 1.0

in phys/module_radiation_driver.F
ZEPSEC (=1.e-12) is supposed to avoid this error, but with an ifort compiler with a debug option it raises the error.

List of modified files:
	modified:   phys/module_radiation_driver.F
---
 phys/module_radiation_driver.F | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/phys/module_radiation_driver.F b/phys/module_radiation_driver.F
index be41215..682af14 100644
--- a/phys/module_radiation_driver.F
+++ b/phys/module_radiation_driver.F
@@ -6863,7 +6863,11 @@ END SUBROUTINE aer_p_int
         zclear = 1.0
         zcloud = 0.0
         DO iz=1,dz
-          zclear = zclear * (1.0-MAX(clfra(iz),zcloud)) / (1.0-MIN(zcloud,1.0-ZEPSEC))
+          if(max(clfra(iz),zcloud).ne.1.0) then
+            zclear = zclear * (1.0-MAX(clfra(iz),zcloud)) / (1.0-MIN(zcloud,1.0-ZEPSEC))
+          else
+            zclear = 0.0
+          endif
           dclt = 1.0 - zclear
           zcloud = clfra(iz)
         END DO
-- 
2.31.1

